src/libcore/str/mod.rs

```diff
@@ -1343,7 +1343,7 @@ generate_pattern_iterators! {
 /// [`str`]: ../../std/primitive.str.html
 #[stable(feature = "rust1", since = "1.0.0")]
 #[derive(Clone, Debug)]
-pub struct Lines<'a>(Map<SplitTerminator<'a, char>, LinesAnyMap>);
+pub struct Lines<'a>(Map<SplitTerminator<'a, char>, fn(&'a str) -> &'a str>);
```

Comment:
Isn't that just a function pointer, which could have been done whenever but intentionally wasn't to avoid dynamic dispatch and a larger object?  It feels like, to use clonable closures, this would be an `impl FnMut` or something.  (I don't know what exactly; maybe it needs abstract type?)